### PR TITLE
Keep window on screen on window start up

### LINKED ISSUE #434: You can tear out windows so that they are off screen
- [x] If you take a tile, and drag it to the bottom right of your desktop, the new window appears entirely off screen.

- [x] i don't think the new windows should appear with it's top left corner at the mouse location - it should be offset.

### PR DESCRIPTION
**Pull request by WPFerg**
Fixes #434

### CODE REVIEW

File: src/window-service.js [hunk 1]
```diff
@@ -203,24 +225,6 @@
...
-            } else {
-                _window.updateOptions({
-                    resizable: true,
-                    minHeight: 510,
-                    minWidth: 918,
-                    maximizable: true
-                });
-            }
-        }
-
```
Comment (owennw) [inline src/window-service.js]: Ha, this must have been a merge fail at one point, good spot!

File: src/window-service.js [hunk 2]
```diff
@@ -177,6 +178,27 @@
...
         }
 
+        snapToScreenBounds(targetWindow) {
+            fin.desktop.System.getMonitorInfo((info) => {
+                targetWindow.getBounds((bounds) => {
+
+                    if (bounds.left < info.virtualScreen.left) {
+                        bounds.left = info.virtualScreen.left;
+                    } else if (bounds.left + bounds.width > info.virtualScreen.right) {
+                        bounds.left = info.virtualScreen.right - bounds.width;
```
Comment (owennw) [inline src/window-service.js]: I take it you can't move `bounds.right` and have that shunt the whole window across?

File: src/window-service.js [hunk 3]
```diff
@@ -177,6 +178,27 @@
...
+                    }
+
+                    if (bounds.top < info.virtualScreen.top) {
+                        bounds.top = info.virtualScreen.top;
```
Comment (owennw) [inline src/window-service.js]: Does this `info(.virtualScreen)` have any info on the taskbar position?

Comment (WPFerg) [inline src/window-service.js]: It does, http://cdn.openfin.co/jsdocs/stable/tutorial-system.getMonitorInfo.html, should I add in taskbar compensation too?

File: src/window-service.js (hunk 2 above)
Comment (WPFerg) [inline src/window-service.js]: There's no `bounds.right`, unfortunately

File: src/window-service.js (hunk 3 above)
Comment (owennw) [inline src/window-service.js]: I think so yes. If the top of the window ended up behind the taskbar it would be unmovable.

Comment (WPFerg) [inline src/window-service.js]: Added. It shrinks the size of the working area.

**Pull request merged**